Jobs.Fetcher.YouTubeStudio/YouTubeGroupsJob.cs

```diff
@@ -0,0 +1,456 @@
+using Serilog.Core;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+
+using Google.Apis.YouTube.v3;
+using Google.Apis.YouTubeAnalytics.v2;
+using Google.Apis.YouTubeAnalytics.v2.Data;
+
+using Andromeda.Common.Jobs;
+using Andromeda.Common.Logging;
+using DataLakeModels;
+using DL = DataLakeModels.Models.YouTube.Studio;
+
+using Jobs.Fetcher.YouTube.Helpers;
+using Jobs.Fetcher.YouTubeStudio.Helpers;
+
+/*
+ * To enable those jobs, you will need to add the following scopes to your list
+ * of authorized scopes:
+
+        https://www.googleapis.com/auth/youtube
+        https://www.googleapis.com/auth/youtubepartner
+
+ * See file utilities/youtube_credentials.py
+ */
+
+namespace Jobs.Fetcher.YouTubeStudio {
+    public abstract class YTSGroupsAbstractJob : AbstractJob {
+        public const string GroupIdentifier = "[JELLYFISH] ";
+        public const int MaxItemsPerGroup = 500;
+        public const int TimeToSleepMs = 1000;
+        protected readonly List<YouTubeService> YTDList;
+        protected readonly List<YouTubeAnalyticsService> YTAList;
+        protected YouTubeService YTD;
+        protected YouTubeAnalyticsService YTA;
+        protected string ChannelId;
+        protected string UploadsListId;
+        public YTSGroupsAbstractJob(
+            List<(YouTubeService, YouTubeAnalyticsService)> services
+            ) {
+            YTDList = new List<YouTubeService>();
+            YTAList = new List<YouTubeAnalyticsService>();
+            foreach (var servicePair in services) {
+                YTDList.Add(servicePair.Item1);
+                YTAList.Add(servicePair.Item2);
+            }
+            Logger = GetLogger();
+        }
+
+        public override void Run() {
+            for (int i = 0; i < YTDList.Count(); i++) {
+                YTD = YTDList[i];
+                YTA = YTAList[i];
+                (ChannelId, UploadsListId) = ApiDataFetcher.FetchChannelInfo(YTD);
+                Logger.Information($"This job is now concerned with channel {ChannelId}");
+                RunBody();
+            }
+        }
+
+        protected abstract void RunBody();
+
+        protected override Logger GetLogger() {
+            return LoggerFactory.GetLogger<DataLakeLoggingContext>(Id());
+        }
+
+        /* API METHODS */
+
+        protected void ReportError(Errors e) {
+            if (e == null) {
+                return;
+            }
+            Logger.Error($@"Response to API request contains an error:
+						Code = {e.Code}
+						Error = {e.Error}
+						ETag = {e.ETag}
+						RequestId = {e.RequestId}
+						");
+            Logger.Error("For more information, see https://googleapis.dev/dotnet/Google.Apis.YouTubeAnalytics.v2/latest/api/Google.Apis.YouTubeAnalytics.v2.Data.Errors.html");
+            throw new Exception("Response to API request contains an error. See message above.");
```

Comment:
This exception is only caught outisde of this job, right? Is this desirable? Should there be error handling inside of the job before leaving it with an error?